### ISSUE TITLE
Generic script to mask the reference genome

### ISSUE THREAD

**Issue by lskatz** [lskatz/lyve-SET#13]
Given start/stop/contig, change certain bases to "N"

**Comment by lskatz** [comment]
Masking could also be done by removing those regions from the bam file.  Either with `bedtools` or with `samtools view`

**Comment by lskatz** [comment]
* `set_findPhages.pl` produces a bed file now.
* Any bed file under `reference/maskedRegions` is read by `launch_set` which produces `reference/maskedRegions.bed` and `unreference/unmaskedRegions.bed`
* `launch_smalt` respects the `unreference/unmaskedRegions.bed` file via `samtools`

**Comment by lskatz** [comment]
TODO: incorporate this into `launch_snap`

**Comment by lskatz** [comment]
It's in `launch_snap` but I noticed that SNAP itself uses the whole defline including the description (after the space in the defline).  Therefore samtools might not identify the correct regions and might produce a blank bam file.  I cannot figure out how to get around this except, I did a regex replace for the description so that it is removed.  However, if the description has special regex characters then it will still produce the zero-byte file.
